### ISSUE TITLE
[Android]

### ISSUE THREAD

**Issue by the2hl** [WhatsApp/stickers#750]
**Specifications**  
Android Version: 10
Device Manufacturer: Samsung Galaxy A40

**Stickers**
- [x] My stickers do comply with all animated sticker requirements

**Problem Description**  
I have downloaded the latest Android demo app version and I've changed  the animated sticker pack with my stickers. I'm getting an *Sticker file might be corrupted or invalid* error when adding the animated sticker pack.
![error-1](https://user-images.githubusercontent.com/26382090/110363142-7986b400-8042-11eb-8b39-0ffdc6801f9f.jpg)
![error-2](https://user-images.githubusercontent.com/26382090/110363155-7db2d180-8042-11eb-820d-8a43f693c876.jpg)
The problem is with the last sticker (Winnie Pooh) because when I do not add it in the content.json, I can add the animated sticker pack with no problem. My question is why am I getting this error when my sticker complies with all requirements?
[winnie_pooh.zip](https://github.com/WhatsApp/stickers/files/6103348/winnie_pooh.zip)

Thank you in advance.

**Comment by imagetag6608** [comment]
same with me. 
I make 10 animation sticker but 4 of them is not able to add in WhatsApp. The other 6 files are working fine on WhatsApp. don't know why it's happening. 
Following validation, 
1) animation sticker 512*512
2) less than 500KB
3) second 0.8 ms to 10 ms
I am following all requirements as listed in the guideline 
Attaching animation sticker file. someone, check and guides me where is my mistake in the animation file.
many developers is facing the same issue in my community. @simonzhexu please answer as soon as possible.
[animation_sticker_not_working.zip](https://github.com/WhatsApp/stickers/files/6134880/animation_sticker_not_working.zip)

**Comment by coroman** [comment]
Hi @imagetag6608 , I had looked your file, i was tried look into your gif file 1.gif, and when i delete 1 frame with 13 ms delay, i can add it to WhatsApp, maybe you should check another gif and try to delete frame with over 10 ms delay.
![1](https://user-images.githubusercontent.com/20947540/111092432-a5d19180-8568-11eb-80c4-b3c5bdbd691b.png)

And here is the proof i can add it to WhatsApp
![image](https://user-images.githubusercontent.com/20947540/111092839-cc43fc80-8569-11eb-8de7-d7accd25d9e3.png)

**Comment by the2hl** [comment]
I wonder how can @imagetag6608 use an animated sticker that does not comply with one of the requirements:

> Animated stickers must have frames with minimum duration of 8ms

**Comment by imagetag6608** [comment]
@the2hl yes i did not understand. i am new in animation things and I am using photoshop frame animation. can you please guide me how can I send 8ms between two frames or explain the thing rule. if yes then its big thank you.

And the rule is 8ms minimum between two frames, so @coroman  image has 3ms delays for every frame. so how its work?

**Comment by the2hl** [comment]
Hi @imagetag6608. I'm also new in animation and I am using GIMP as image editor. In GIMP, you can set the frame duration between parenthesis: (XX ms)
![image](https://user-images.githubusercontent.com/26382090/111354020-29cd6b80-8686-11eb-8146-5d81cfa98667.png)
You can check this tutorial: https://www.linuxtopia.org/online_books/graphics_tools/gimp_advanced_guide/gimp_guide_node80_001.html 
According to the requirements (README.md), you should set at least a 8ms frame duration. I don't know why @imagetag6608 has been able to add your sticker having frames with 3ms duration.

**Comment by coroman** [comment]
> @the2hl yes i did not understand. i am new in animation things and I am using photoshop frame animation. can you please guide me how can I send 8ms between two frames or explain the thing rule. if yes then its big thank you.
> 
> And the rule is 8ms minimum between two frames, so @coroman image has 3ms delays for every frame. so how its work?
> 
> I know its all every basic question but I did not have proper knowledge so sorry for that.

@imagetag6608 There is one frame with 13ms delay when i looked up into

**Comment by BattleShipPark** [comment]
> Hi @imagetag6608 , I had looked your file, i was tried look into your gif file 1.gif, and when i delete 1 frame with 13 ms delay, i can add it to WhatsApp, maybe you should check another gif and try to delete frame with over 10 ms delay.
I can't understand why pack is available after deleting a frame with over 10ms delay. Anyone or @coroman knows it? I could not find it from requirements.
